contributor-guide/modules/ROOT/pages/testing/static-analysis.adoc

```diff
@@ -0,0 +1,56 @@
+= Static Analysis
+:navtitle: Static Analysis
+
+Static analysis is a method of debugging by examining source code before a program is run. It's done by analyzing a set of code against a set (or multiple sets) of coding rules. This type of analysis addresses weaknesses in source code that might lead to vulnerabilities. The objective is to find errors, bugs, or deviations from best practices early in the development process to improve code quality and maintainability.
+
+Here are a few key characteristics of static analysis:
+
+* Static analysis is performed without actually executing the code. It reviews the code in its non-runtime, static state.
+
+* Many static analysis methods are automated. Automated tools can quickly scan large codebases and identify potential issues based on predefined rules or patterns.
+
+* Static analysis tools can detect a variety of issues, including syntax errors, type safety issues, race conditions, unused code, memory leaks, and other potential vulnerabilities.
+
+* Static analysis can help ensure that code adheres to coding standards and guidelines.
+
+* Besides looking for bugs or vulnerabilities, static analysis tools often compute metrics that give an overall impression of code quality, like cyclomatic complexity, code duplication, or code comments ratio.
+
+Because it's performed early in the software development lifecycle - often as part of an integrated development environment (IDE) or CI pipeline - static analysis helps developers catch and fix issues early, which can save time and resources. Refer to the xref:testing/continuous-integration.adoc#_coverity_scan[Coverity Scan] documentation.
+
+== Treat Warnings as Errors
+
+One of the most important things for library developers to do is to turn on compiler settings to treat Warnings as Errors. In practice, some of these warnings will need to be whitelisted as they will be false positives. Otherwise, most warnings should be addressed as errors.
```

Comment:
Please mention the `warnings-as-errors=on` feature in B2. While there is a [`CMAKE_COMPILE_WARNING_AS_ERROR`](https://cmake.org/cmake/help/latest/variable/CMAKE_COMPILE_WARNING_AS_ERROR.html#variable:CMAKE_COMPILE_WARNING_AS_ERROR), it's 3.24 >= so may not be an option

Reply:
ok